src/views/github/index.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { useState } from "react";
+import { Redirect } from 'react-router-dom';
+
+import { parseOAuthCode } from './../../actions';
+
+
+// TODO: use state for verification + dont redirect
+export const OAuthCallbackView = () => {
+  const [code, setCode] = useState<any>("")
+  
+
+  // TODO: is this something that can be initialized?
+  //       or does this need to be implemented better?
+  if (code === "") {
+    setCode(parseOAuthCode());
+  }

```

Comment:
seems like `code` will always be "" anyway, why do we need to check?
i think whats happening here is
- on load :setCode(parseOAuthCode());
- then before the return statement you can check if code is "" or not

Reply:
Yea I think this is partially me just struggling with how the hook works and react manages state. I'll keep working on playing with this sort of stuff. I think for now maybe we tradeoff this ugliness for the merge. That way someone more knowledgeable might be able to come swoop in and make the tweak.

Reply:
nah for this one, we need to fix it. the logic doesnt seem right to me 😅

Reply:
So `setCode(newCode);` in my eyes would change the state to the code if it exists. I'm not too familiar with what's going on under the hood, but if the state changes but the redirect fails I wouldn't want to run `parseOAuthCode` again. Maybe that's overthinking it though.

Reply:
but currently it's setting if code is empty, which will always run anyway, bcos the default state of `code` is ""
so what we can do is just remove the if clause and add a condition in return statement to only render it if `code` is not empty which means we have run the `parseOAuthCode`
unless im missing something, like what is redirect, i dont see it in this file?

Reply:
`<Redirect ... />` but I'm gunna test this out because you're probably right.

Reply:
Looks like it works!